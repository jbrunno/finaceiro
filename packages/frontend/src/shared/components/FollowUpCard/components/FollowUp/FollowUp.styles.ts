import { Grid, TimelineContent, Typography } from '@frontend/components';
import { css, styled } from '@frontend/styles';

export const GridStyled = styled(Grid)`
  max-height: ${(props) => props.theme.fn.size(250)};
  overflow-y: scroll;
`;

export const TimelineContentStyled = styled(TimelineContent)`
  ${({ theme }) => css`
    padding: ${theme.fn.spacing(0, 24)};
    margin-bottom: ${theme.fn.spacing(24)};
  `}
`;

export const TypographyStyled = styled(Typography)`
  margin: ${(props) => props.theme.fn.spacing(8, 0, 16)};
`;
