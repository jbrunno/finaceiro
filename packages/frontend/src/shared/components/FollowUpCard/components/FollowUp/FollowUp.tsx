import {
  Timeline,
  TimelineConnector,
  TimelineDot,
  TimelineItem,
  TimelineSeparator,
  Typography,
} from '@frontend/components';
import { timelineItemClasses } from '@mui/lab';
import { formatDateWithHours } from '@frontend/utils';
import {
  GridStyled,
  TimelineContentStyled,
  TypographyStyled,
} from './FollowUp.styles';
import { ClienteFollowUps } from '../../FollowUpCard.gql';

type FollowUpProps = {
  items: ClienteFollowUps[] | undefined;
};

export function FollowUp({ items }: FollowUpProps) {
  return (
    <GridStyled>
      <Timeline
        sx={{
          [`& .${timelineItemClasses.root}:before`]: {
            flex: 0,
            padding: 0,
          },
          padding: 0,
        }}
      >
        {items?.map((item) => (
          <TimelineItem key={Math.random().toString()}>
            <TimelineSeparator>
              <TimelineDot color="secondary" />
              <TimelineConnector />
            </TimelineSeparator>
            <TimelineContentStyled>
              <Typography variant="caption" color="textPrimary">
                {formatDateWithHours(new Date(item.dataRegistro))}
              </Typography>
              <TypographyStyled variant="subtitle1" color="textPrimary">
                {item.situacao.nome}
              </TypographyStyled>

              <Typography variant="caption" color="textPrimary">
                Nome do Operador
              </Typography>
              <TypographyStyled color="textPrimary">
                {item.usuario.nome}
              </TypographyStyled>

              <Typography variant="caption" color="textPrimary">
                Descrição
              </Typography>
              <TypographyStyled color="textPrimary">
                {item.descricao}
              </TypographyStyled>
            </TimelineContentStyled>
          </TimelineItem>
        ))}
      </Timeline>
    </GridStyled>
  );
}
