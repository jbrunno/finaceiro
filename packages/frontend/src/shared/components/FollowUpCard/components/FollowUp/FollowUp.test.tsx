import { render } from '@test-utils/component-testing';
import { FollowUp } from './FollowUp';

test('should render FollowUp', () => {
  const { container } = render(<FollowUp items={[]} />);
  expect(container).toMatchSnapshot();
});
