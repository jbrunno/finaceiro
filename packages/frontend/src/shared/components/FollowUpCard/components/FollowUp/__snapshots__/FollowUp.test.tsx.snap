// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FollowUp 1`] = `
.emotion-1 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-height: calc(10.9375rem + 75px);
  overflow-y: scroll;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px 16px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 0rem;
}

.emotion-2 .MuiTimelineItem-root:before {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  padding: 0rem;
}

<div>
  <div
    class="MuiGrid-root emotion-0 emotion-1"
  >
    <ul
      class="MuiTimeline-root MuiTimeline-positionRight emotion-2"
    />
  </div>
</div>
`;
